tox.ini

```diff
@@ -31,8 +31,10 @@ description = lint with pylint
 skip_install = true
 skipsdist = true
 deps = -r requirements-dev.txt
+allowlist_externals = make
 commands =
     {envpython} -m pylint --load-plugins pylint_pydantic src/instructlab/ tests/
+    make toml-lint
```

Comment:
I would make a separate tox env for toml linting `[testenv:tomllint]`. This block is for pylint. 

```ini
[testenv:tomllint]
description = lint pyproject.toml
skip_install = true
skipsdist = true
deps =
commands =
    make toml-lint
    make toml-fmt
    sh -c 'git diff --exit-code || (echo "pyproject.toml formatting is incorrect. Please run \"make toml-fmt\" and commit the changes." && exit 1)'
allowlist_externals = make
```

Then the work flow step will call `tox -e tomllint`.

Reply:
Note: this code will lint AND format. That's why I've put the `make toml-lint` in the `lint` tox env. I think it makes sense to have all the lining happening under a single tox env and the same for the fmt (like ruff).

What do you think?

Reply:
> this code will lint AND format.

Yes, but this is also what the ruff script does. First it lints and then it formats and fails if the formatting makes a change. So doing the same for pyproject.toml is consistent.

Reply:
Ok!